docs/start/api.tx.wrap.md

```diff
@@ -0,0 +1,44 @@
+# Complex transactions
+
+Up till now we have focussed on the base operation of transactions. There are however some more complex operations that deserve some more information, for instance when doing either democracy proposals or excuting sudo calls, in both these cases the transaction send a call or proposal to be evaluated.
+
+## Sudo use
+
+When running a development chain (Polkadot/Substrate with a `--dev` flag), or in certain testnets a sudo module is available - just like the sudo command on *nix systems, it allows root-level access to perform actions. For instance, we can perform a `setBalance` on an account -
+
+```js
+...
+// get the  current sudo key in the system
+const sudoKey = await api.query.sudo.key();
+
+// lookup from keyring (assuming we have added all, on --dev this would be `//Alice`)
+const sudoPair = keyring.getPair(sudoKey);
+
+// send the actual sudo transaction
+const unsub = await api.tx.sudo
+  .sudo(
+    api.tx.balances.setBalance(ADDR, 12345, 0)
+  )
+  .signAndSend(sudoPair, (result) => { ... });
+```
+
+The above is really quite straight-forward, the `sudo.sudo(<call>)` call takes 1 parameter, which is a `Call`. We construct this via the `api.tx` and pass it through. The only difference is that this nested call has no actual `.signAndSend` on it, rather it is only used as a container for data.
+
+Exactly the same would apply to the standard `democracy.propose(<proposal>, <value>)`, for instance we can just swap the above sudo wrapper with a proposal and add the correct fees for the proposal.
+
+## Complex types
+
+As indicated in previous sections (we will cover types in more detail next), the API will format the inputs into the actual type required for submission. For primitives such as numbers, this is quite understandable, but it is worth spending at least one example on cases where an object is provides as an input. For instance, making a call to validate -
+
+```js
+...
+const txHash = await api.tx.staking.validate({
+  validatorPayment: 12345
+});
+```
+
+In the above example, all we need to provide is a call with the fields for the `ValidatorPrefs` object. (Any fields not defined will be set to the default for that type, i.e. all zero).
```

Comment:
I think this would benefit from a link to the actual struct that you're sending. https://github.com/paritytech/substrate/blob/master/srml/staking/src/lib.rs#L346-L354

Someone who's developing their dApp frontend, but doesn't know all the internals of staking on Polkadot or Substrate may be confused about where `validatorPayment` came from.

Alternatively you could show a sample rust struct and the corresponding call in javascript.

Reply:
Made a claifying comment.